lib/puppet/indirector/yaml.rb

```diff
@@ -19,17 +19,20 @@ def find(request)
   def save(request)
     raise ArgumentError.new("You can only save objects that respond to :name") unless request.instance.respond_to?(:name)
 
-    file = path(request.key)
+    files = path(request.key)
```

Comment:
I reworked so it so it's more contained to the Puppet::Indirector::Report::Yaml class, and have staged that for further review as #4914.